### PR TITLE
Ccp 1231 heritage foods improvements

### PR DESCRIPTION
**Pull request by DannyShopgate**
# CCP-1231 Terms and Conditions per product

## Description

Added support for specific product Id configurations. Terms and Conditions can be applied to specific products.

## To Test
Used shop number: 31113 and added products titled Test Product 001 and Test Product 002 that can be used with default extension configuration. 

## Type of change
- [x] New feature

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I updated the CHANGELOG.md

### CODE REVIEW

File: frontend/config.json [hunk 1]
```diff
@@ -1,4 +1,40 @@
 {
-  "checkBoxValues": [],
+  "checkboxValues": [
```
[the remark this answers is not in the document]
Comment (aaron-martin) [reply frontend/config.json]: done

File: extension-config.json [hunk 2]
```diff
@@ -26,14 +26,50 @@
     }
   ],
   "configuration": {
-    "checkBoxValues": {
+    "checkboxValues": {
       "type": "admin",
       "destination": "frontend",
-      "default": [],
+      "default": [
```
[the remark this answers is not in the document]
Comment (aaron-martin) [reply extension-config.json]: done

File: frontend/portals/CartItemListAfter/index.jsx [hunk 3]
```diff
@@ -1,12 +1,12 @@
 import React from 'react';
-import CheckBoxes from '../../components/CheckBoxes';
+import Checkboxes from '../../components/Checkboxes';
```
[the remark this answers is not in the document]
Comment (aaron-martin) [reply frontend/portals/CartItemListAfter/index.jsx]: done

File: frontend/components/Checkboxes/index.jsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+   */
+  render() {
+    const {
+      termsToDisplay,
+    } = this.props;
+    const {
+      checkValues,
+    } = this.state;
+    const checkboxes = termsToDisplay.map((product, index) => {
+      if (typeof checkValues[index] !== 'undefined') {
```
Comment (aaron-martin) [inline frontend/components/Checkboxes/index.jsx]: I think you should test if typeof checkValues[index] === 'undefined' and return null in that case and take the non-error state out of an if statement.

File: frontend/selectors/index.js [hunk 5]
```diff
@@ -1,12 +1,83 @@
...
+
+/**
+ * @param {Object} state state
+ * @returns {Object}
+ */
+const getExtensionsState = state => state.extensions;
+
+const getExtensionState = createSelector(
+  getExtensionsState,
+  extensionsState => extensionsState['@shopgate/terms-and-conditions/isOrderableByTerms']
```
Comment (aaron-martin) [inline frontend/selectors/index.js]: I think you should set a constant for @shopgate/terms-and-conditions/isOrderableByTerms

File: frontend/selectors/index.js [hunk 6]
```diff
@@ -1,12 +1,83 @@
...
 );
+
+export const getTermsToDisplay = createSelector(
+  getTermProductIds,
+  (productIds) => {
+    if (!productIds) {
+      return null;
+    }
+    return checkboxValues.filter((value =>
+      productIds.includes(value.displayOn) || value.displayOn === 'all products'));
```
Comment (aaron-martin) [inline frontend/selectors/index.js]: I think a constant should be used for 'all products'

**Comment by aaron-martin** [comment]
I think it should be possible to add a variant ID to the configuration to have variant product specific checkboxes be shown. I know this is more of an edge case but supporting this behavior would be simple and would make the extension more flexible.

File: frontend/components/Checkboxes/index.jsx (hunk 4 above)
Comment (DannyShopgate) [reply frontend/components/Checkboxes/index.jsx]: updated

File: frontend/selectors/index.js (hunk 5 above)
Comment (DannyShopgate) [reply frontend/selectors/index.js]: updated

File: frontend/selectors/index.js (hunk 6 above)
Comment (DannyShopgate) [reply frontend/selectors/index.js]: Updated

**Pull request merged**